lib/render/FlowRenderer.cpp

```diff
@@ -450,13 +454,26 @@ FlowRenderer::_updateFlowCacheAndStates( const FlowParams* params )
     // If names not the same, entire stream is out of date
     // Note: variable names are kept in VaporFields.
     std::vector<std::string> varnames = params->GetFieldVariableNames();
+    for( const auto& e : varnames )
```

Comment:
I'm not sure that this code does what we want. It should not be an error for a user to select fewer than three variables. The missing variables should be treated as the "zero" variable: a variable that contains all zeros. A common use case is to do flow advection using only the horizontal components of velocity and ignore the vertical.

Reply:
Oh really? That's interesting, and I I'll change the behavior of it.